### PR TITLE
Added crossplane-azure-upbound Helm chart

### PR DESCRIPTION
**Pull request by paolosalvatori**
I added a new Helm chat in the `crossplane-azure-upbound` folder to install the Crossplane upbound packages for Azure. The `ProviderConfig` CRD supports both the service principal and managed identity credentials.

### CODE REVIEW

File: charts/aws-ebs-csi/Chart.yaml [hunk 1]
```diff
@@ -15,10 +15,10 @@ type: application
 # This is the chart version. This version number should be incremented each time you make changes
 # to the chart and its templates, including the app version.
 # Versions are expected to follow Semantic Versioning (https://semver.org/)
-version: 1.0.0
+version: 2.0.1
```
Comment (csantanapr) [inline charts/aws-ebs-csi/Chart.yaml:18]: Do not modify this file.

File: charts/aws-fargate-fluentbit/Chart.yaml [hunk 2]
```diff
@@ -15,10 +15,10 @@ type: application
 # This is the chart version. This version number should be incremented each time you make changes
 # to the chart and its templates, including the app version.
 # Versions are expected to follow Semantic Versioning (https://semver.org/)
-version: 1.0.0
+version: 2.0.1
```
Comment (csantanapr) [inline charts/aws-fargate-fluentbit/Chart.yaml:18]: Do not modify this file.

File: charts/crossplane-aws-upbound/Chart.yaml [hunk 3]
```diff
@@ -17,6 +17,7 @@ type: application
 # Versions are expected to follow Semantic Versioning (https://semver.org/)
 version: 3.0.0
 
+
```
Comment (csantanapr) [inline charts/crossplane-aws-upbound/Chart.yaml]: Do not modify this file.

File: charts/crossplane-aws-upbound/values.yaml [hunk 4]
```diff
@@ -1,30 +1,7 @@
 global:
   enabled_aws_upbound: true
 
-deploymentRuntimeConfig:
```
Comment (csantanapr) [inline charts/crossplane-aws-upbound/values.yaml:4]: Do not modify this file. Do not delete content

File: charts/crossplane-aws/Chart.yaml [hunk 5]
```diff
@@ -17,6 +17,7 @@ type: application
 # Versions are expected to follow Semantic Versioning (https://semver.org/)
 version: 2.2.0
 
+
```
Comment (csantanapr) [inline charts/crossplane-aws/Chart.yaml]: Do not modify this file.

File: charts/teams/Chart.yaml [hunk 6]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v2
 name: team
 description: A Helm Chart to bootstrap a Namespace in a Multi-Tenancy setup
-version: 1.0.0
+version: 1.0.1
```
Comment (csantanapr) [inline charts/teams/Chart.yaml:4]: Do not modify this file.

**Review by csantanapr — CHANGES_REQUESTED** [review]
Thanks for the PR !
Some requested changes:
- Please do not modified files unrelated to new helm chart
- Signoff DCO

File: charts/crossplane-azure-upbound/values.yaml [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+  enabled_azure_upbound: true
+
+deploymentRuntimeConfig:
+  enabled: false
+  metadata:
+    name: "upbound-azure-runtime-config"
+    role_arn: ""
+    annotations: {}
+    labels:
+      app.kubernetes.io/managed-by: Helm
```
Comment (csantanapr) [inline charts/crossplane-azure-upbound/values.yaml]:
```suggestion
    labels: {}
```

The label `app.kubernetes.io/managed-by: Helm` is automatically put by helm cli, you don't want this present when using argocd

File: charts/crossplane-azure-upbound/values.yaml [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+        annotations: {}
+        labels:
+        name: azure-provider
+
+provider:
+  enabled: true
+  metadata:
+    annotations: {}
+    labels:
+      app.kubernetes.io/managed-by: Helm
```
Comment (csantanapr) [inline charts/crossplane-azure-upbound/values.yaml]:
```suggestion
    labels: {}
```

The label `app.kubernetes.io/managed-by: Helm` is automatically put by helm cli, you don't want this present when using argocd

File: charts/crossplane-azure-upbound/values.yaml [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+    registry: xpkg.upbound.io/upbound
+    version: v0.42.0
+
+providerConfig:
+  enabled: true
+  metadata:
+    name: default
+    annotations: {}
+    labels:
+      app.kubernetes.io/managed-by: Helm
```
Comment (csantanapr) [inline charts/crossplane-azure-upbound/values.yaml]:
```suggestion
    labels: {}
```

The label `app.kubernetes.io/managed-by: Helm` is automatically put by helm cli, you don't want this present when using argocd

File: charts/crossplane-azure-upbound/values.yaml [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+  spec:
+    credentials:
+      source: Secret
+      secretRef:
+        namespace: crossplane-system
+        name: azure-secret
+        key: creds
+    
+providers:
```
Comment (csantanapr) [inline charts/crossplane-azure-upbound/values.yaml]: Is this a complete list?

File: charts/crossplane-azure-upbound/templates/provider-config.yaml [hunk 11]
```diff
@@ -0,0 +1,36 @@
+{{- if and .Values.global.enabled_azure_upbound .Values.providerConfig.enabled }}
+{{- with .Values.providerConfig }}
+apiVersion: azure.upbound.io/v1beta1
+kind: ProviderConfig
+metadata:
+  name: {{ .metadata.name }}   
+  annotations:
+    argocd.argoproj.io/sync-wave: "4"
+    argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
+    helm.sh/hook: post-install
```
Comment (csantanapr) [inline charts/crossplane-azure-upbound/templates/provider-config.yaml:10]: I really like this, we should do it for the other providers.

File: charts/crossplane-azure-upbound/templates/providers.yaml [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+    {{ $key }}: {{ $value | quote }}
+    {{- end }}
+  {{- if $.Values.provider.metadata.labels }}
+  labels:
+    {{- range $key, $value := $.Values.provider.metadata.labels }}
+    {{ $key }}: {{ $value | quote }}
+    {{- end }}
+  {{- end }}
+spec:
+  package: {{ $.Values.provider.package.registry }}/provider-azure-{{ . }}:{{ $.Values.provider.package.version }}
```
Comment (csantanapr) [inline charts/crossplane-azure-upbound/templates/providers.yaml]:
```suggestion
  package: {{ $.Values.provider.package.registry }}/provider-azure-{{ . }}:{{ $.Values.provider.package.version }}
  runtimeConfigRef:
    apiVersion: pkg.crossplane.io/v1beta1
    kind: DeploymentRuntimeConfig
    name: {{ $.Values.deploymentRuntimeConfig.metadata.name }}
```

You are missing the reference to the deployment runtime config that you are creating

**Review by csantanapr — CHANGES_REQUESTED** [review]
I took a look at your chart and found a few things to address

File: charts/crossplane-azure-upbound/values.yaml (hunk 10 above)
Comment (paolosalvatori) [reply charts/crossplane-azure-upbound/values.yaml]: Yes, it is

**Pull request closed without merging**